File: Python/Arrays/reverse_string.py
import sys

def reverse_string(string):
    """
        Reverse provided string
    """ 
    # Check for invalid input
    if string is None or type(string) is not str:
        return 'Invalid Input'

    length = len(string)
    
    # Check for single character strings
    if length < 2:
        return string # Return string itself

    # Create empty string to store reversed string
    reversed_string = ""
    for i in range(length-1, -1, -1):
        reversed_string += string[i]
    return reversed_string



print(reverse_string(sys.argv[1]))

File: Javascript/Arrays/reverse_string.js
let reverse_string = (string) =>{

    // Check if input is valid
    if (!string || typeof(string) !== 'string'){
        return 'Invalid Input';
    }

    if (string.length < 2){
        return string;
    }

    // Create empty string to store reversed string
    let reversed_string = "";
    // Loop through provided string backwards and add each element to 'reversed_string'
    for( let i=string.length -1; i >= 0; i--){
        reversed_string += string[i];
    }
    return reversed_string;
}

// Receive command line argument
// const input_string = process.argv[2]

const input_string = 'Hello! My name is Devesh.'
console.log(reverse_string(input_string))